api/src/main/java/org/openmrs/module/reporting/report/renderer/JSONReportRenderer.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.reporting.report.renderer;
+
+import org.openmrs.Cohort;
+import org.openmrs.annotation.Handler;
+import org.openmrs.module.reporting.common.Localized;
+import org.openmrs.module.reporting.dataset.DataSet;
+import org.openmrs.module.reporting.dataset.DataSetColumn;
+import org.openmrs.module.reporting.dataset.DataSetRow;
+import org.openmrs.module.reporting.report.ReportData;
+import org.openmrs.module.reporting.report.ReportRequest;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.List;
+
+/**
+ * ReportRenderer that renders to a default XML format
+ */
+@Handler
+@Localized("reporting.JSONTemplateRenderer")
+public class JSONReportRenderer extends ReportDesignRenderer {
+
+	/**
+	 * @see ReportRenderer#getFilename(ReportRequest)
+	 */
+    @Override
+	public String getFilename(ReportRequest request) {
+		return getFilenameBase(request) + ".json";
+	}
+
+	/**
+	 * @see ReportRenderer#getRenderedContentType(ReportRequest)
+	 */
+	public String getRenderedContentType(ReportRequest request) {
+		return "text/json";
+	}
+
+	/**
+	 * @see ReportRenderer#render(ReportData, String, OutputStream)
+	 */
+//	Generates JSON file
+	public void render(ReportData results, String argument, OutputStream out) throws IOException, RenderingException {
+
+		Writer w = new OutputStreamWriter(out, "UTF-8");
```

Comment:
What specifications are you using to generate this JSON?  A few thoughts:
* The reportingrest module already supports a web service that returning a JSON representation of a data set.  Is there a good reason not to try to keep this consistent with each other?
* Generally defaulting to a "toString" representation of an object is not going to be what you want to do.  What if the data values are OpenMRS objects?  These should probably be returned with a reference to their UUID.  There are likely other examples (what format do we want for Dates, etc).
* If you _do_ just want a toString representation, I'd recommend using the ObjectUtils.format method in the reporting module.
Obviously you can also put this renderer directly into your implementation module if you want it to do this type of format specifically.  This does not have to live in the reporting module for it to work.